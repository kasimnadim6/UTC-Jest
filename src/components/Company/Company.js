import React from 'react';
import { useState } from 'react';

const Company = ({ name }) => {
  const [country, setCountry] = useState('');
  const inputHandler = (e) => {
    setCountry(e.target.value);
  };
  return (
    <div data-testid="companyNameEL">
      Company name is: {name}
      <h2>Welcome to my Company's official website</h2>
      <p data-testid="employee-count" style={{ color: 'green' }}>
        Employee count 3,18,400
      </p>
      <input
        type="text"
        placeholder="Enter country"
        value={country}
        onChange={inputHandler}
      />
      <button className="btn" role="join-cognizant">
        Join
      </button>
    </div>
  );
};

export default Company;
