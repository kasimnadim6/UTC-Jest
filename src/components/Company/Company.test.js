import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Company from './Company';

describe(`Company component`, () => {
  it('should be available within app', () => {
    expect(Company).toBeDefined();
  });

  it('should receive props and display it', () => {
    render(<Company name="CTS" />);
    const companyNameEL = screen.getByTestId('companyNameEL');
    expect(companyNameEL).toHaveTextContent('CTS');
  });
  it("has Welcome to My Company's official website", () => {
    render(<Company />);
    const officialTextEL = screen.getByText(
      "Welcome to my Company's official website"
    );
    expect(officialTextEL).toBeInTheDocument();
  });
  it('should have text with color green', () => {
    render(<Company />);
    expect(screen.getByTestId('employee-count')).toHaveStyle('color: green');
  });
  it("has input element with placeholder 'Enter country'", () => {
    render(<Company />);
    expect(screen.getByPlaceholderText('Enter country')).toBeTruthy();
  });
  it('has button element with btn class', () => {
    render(<Company />);
    const btnEl = screen.getByRole('join-cognizant');
    expect(btnEl).toHaveClass('btn');
  });
  it('should update country on onChange event', () => {
    render(<Company />);
    const countryInput = screen.getByPlaceholderText('Enter country');
    fireEvent.change(countryInput, { target: { value: 'India' } });
    expect(countryInput.value).toBe('India');
  });

  it('should have right Company component Snapshot', () => {
    const snapshotJSON = renderer.create(<Company name="Cognizant" />);
    expect(snapshotJSON).toMatchSnapshot();
  });
});
