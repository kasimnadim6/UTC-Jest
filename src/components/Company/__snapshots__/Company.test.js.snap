// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company component should have right Company component Snapshot 1`] = `
<div
  data-testid="companyNameEL"
>
  Company name is: 
  Cognizant
  <h2
    style={
      Object {
        "color": " #61dafb",
      }
    }
  >
    Welcome to my Company's official website
  </h2>
  <p
    data-testid="employee-count"
    style={
      Object {
        "color": "green",
      }
    }
  >
    Employee count 3,18,400
  </p>
  <input
    onChange={[Function]}
    placeholder="Enter country"
    type="text"
    value=""
  />
  <button
    className="btn"
    role="join-cognizant"
  >
    Join
  </button>
</div>
`;
